GenumKit/Parsers/FontsFileParser.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  FontsFileParser.swift
+//  Pods
+//
+//  Created by Derek Ostrander on 3/7/16.
+//
+//
+
+import Foundation
+import AppKit.NSFont
+
+typealias Font = (name: String, variant: String)
+
+extension CFString {
+    var alphanumericString: String {
+         return (self as String).componentsSeparatedByCharactersInSet(NSCharacterSet.alphanumericCharacterSet().invertedSet).joinWithSeparator("")
+    }
+}
+
+extension CTFont {
+    static func loadFont(path: String) -> Font? {
+        let inData = NSData(contentsOfFile: path)
+        var error: Unmanaged<CFError>?
+        guard
+            let provider: CGDataProvider = CGDataProviderCreateWithCFData(inData),
+            let font = CGFontCreateWithDataProvider(provider)
+            where CTFontManagerRegisterGraphicsFont(font, &error) else {
+                return nil
+        }
+        return CTFontCreateWithGraphicsFont(font, 0, nil, nil).font
+    }
+
+    private var font: Font? {
+        let family = CTFontCopyFamilyName(self).alphanumericString
+        let name = CTFontCopyFullName(self).alphanumericString.stringByReplacingOccurrencesOfString(family, withString: "")
+        if family.characters.count == 0 {
+            print("There was a problem detecting the name of the font. \(CTFontCopyFamilyName(self)) -- \(CTFontCopyFullName(self)) ")
+            return nil
+        }
+        return (family, name.characters.count > 0 ? name : "Regular")
+    }
+}
+
+// MARK: Interface
+public final class FontsFileParser {
+    public var entries: [String: Set<String>] = [:]
+
+    public init() {}
+
+    public func parseFonts(path: String) {
+        for file in contentsOfDirectory(path) {
+            let fullPath = (path as NSString).stringByAppendingPathComponent(file)
```

Comment:
Yeah that's pretty powerful.

I just realized that, as you'll then need to filter the enumerated files to only process font files (and skip any file in the scanned directory that might be present but not be a font file, in case the user scan for example the whole `$PROJECT_DIR`), that you might even use this method of `NSFileManager` instead which is even more powerful:

```
let dirEnum = NSFileManager.defaultManager().enumeratorAtURL(
  NSURL(fileURLWithPath: path),
  includingPropertiesForKeys:[NSURLTypeIdentifierKey],
  options: [.SkipsPackageDescendants, .SkipsHiddenFiles]
  errorHandler: nil)
```

This will enumerate the files recursively and I guess you should then be able to check `dirEnum.fileAttributes[NSURLTypeIdentifierKey]` and compare it with `public.font` to only process font files?